File: LC/3541/3541.py
# link of question: https://leetcode.com/problems/find-most-frequent-vowel-and-consonant/
def maxFreqSum(self, s: str) -> int:
    vowel = {}
    consonant = {}
    for i in s:
        if i in 'aeiou':
            if i in vowel:
                vowel[i] += 1
            else:
                vowel[i] = 1
        else:
            if i in consonant:
                consonant[i] += 1
            else:
                consonant[i] = 1

    max_vowel_value = max(vowel.values()) if vowel else 0
    max_consonant_value = max(consonant.values()) if consonant else 0
    return max_vowel_value + max_consonant_value

File: online_judge/LC/3541/3541.cpp
// link of question: https://leetcode.com/problems/find-most-frequent-vowel-and-consonant/
class Solution {
public:
    int maxFreqSum(string s) {
        vector<pair<char, int>> vowel;
        vector<pair<char, int>> consonant;
        bool is_find = false;
        for (int i = 0; i < s.length(); i++) {
            if (s[i] == 'a' or s[i] == 'e' or s[i] == 'i' or s[i] == 'o' or s[i] == 'u') {
                for (int j = 0; j < vowel.size(); j++) {
                    if (s[i] == vowel[j].first) {
                        vowel[j].second++;
                        is_find = true;
                        break;
                    }
                }
                if (is_find) {
                    is_find = false;
                } else {
                    vowel.push_back({s[i], 1});
                }
            } else {
                for (int j = 0; j < consonant.size(); j++) {
                    if (s[i] == consonant[j].first) {
                        consonant[j].second++;
                        is_find = true;
                        break;
                    }
                }
                if (is_find) {
                    is_find = false;
                } else {
                    consonant.push_back({s[i], 1});
                }
            }
        }

        int max_vowel_value = 0;
        for (int i = 0; i < vowel.size(); i++) {
            if (max_vowel_value < vowel[i].second) {
                max_vowel_value = vowel[i].second;
            }
        }

        int max_consonant_value = 0;
        for (int i = 0; i < consonant.size(); i++) {
            if (max_consonant_value < consonant[i].second) {
                max_consonant_value = consonant[i].second;
            }
        }

        return max_vowel_value + max_consonant_value;
    }
};